### PR TITLE
expression: refactor functions.

### PR DESCRIPTION
**Pull request by hanfei1991**
one step of refactor.
@shenli @coocood @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: expression/builtin_compare.go [hunk 1]
```diff
@@ -225,63 +213,57 @@ type builtinCompareSig struct {
...
-		default:
-			return d, errInvalidOperation.Gen("invalid op %v in comparison operation", op)
-		}
-		if result {
-			d.SetInt64(oneI64)
-		} else {
-			d.SetInt64(zeroI64)
+	}
+	if a.IsNull() || b.IsNull() {
+		// for <=>, if a and b are both nil, return true.
```
Comment (zimulala) [inline expression/builtin_compare.go]: s/for/For

File: expression/builtin_compare.go [hunk 2]
```diff
@@ -225,63 +213,57 @@ type builtinCompareSig struct {
...
+		// if a or b is nil, return false.
```
Comment (zimulala) [inline expression/builtin_compare.go]: s/if/If

**Comment by zimulala** [comment]
Rest LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@zimulala @coocood PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**